drivers/gles3/storage/texture_storage.cpp

```diff
@@ -1130,11 +1130,17 @@ void TextureStorage::_update_render_target(RenderTarget *rt) {
 		rt->fbo = system_fbo;
 		return;
 	}
-
-	rt->color_internal_format = rt->is_transparent ? GL_RGBA8 : GL_RGB10_A2;
-	rt->color_format = GL_RGBA;
-	rt->color_type = rt->is_transparent ? GL_BYTE : GL_UNSIGNED_INT_2_10_10_10_REV;
-	rt->image_format = Image::FORMAT_RGBA8;
+	if (rt->force_high_precision) {
+		rt->color_internal_format = rt->is_transparent ? GL_RGBA16F : GL_RGB16F;
```

Comment:
Mobile devices don't typically support RGBA16 or RGB16 (some do through extensions though). So we can't rely on support being available. We will have to add a config setting ``framebuffer_half_float_supported`` and then fall back to low precision if it is false. You can initialize the config the same way we do in 3.x, e.g. in storage/config.cpp:
```
#ifdef GLES_OVER_GL
	config.framebuffer_float_supported = true;
#else
	framebuffer_float_supported = config.extensions.has("GL_EXT_color_buffer_float");
#endif

Reply:
Had to do a big rebase, everything looks fine but hopefully i didn't missed something :crossed_fingers:
Would'nt it be better to just use the `#ifdef` to ignore the force_high_precision ? Since the high precision for thoses devices is actually the default behavior.

Reply:
Added a check for the support of float framebuffer